### PR TITLE
Add upgrade for field descriptor visibility flag

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Related issues/PRs | #3895, #3896
| License | MIT

#### What's in this PR?

Add upgrade for field descriptor visibility flag.

#### Why?

Howto update field descriptors to sulu 2.0

#### To Do

- [x] Create a documentation PR

### CODE REVIEW

File: src/Sulu/Component/Rest/ListBuilder/Doctrine/FieldDescriptor/DoctrineFieldDescriptor.php [hunk 1]
```diff
@@ -117,7 +117,7 @@ public function getJoins()
     /**
      * {@inheritdoc}
      */
-    public function compare(FieldDescriptorInterface $other)
+    public function compare(self $other)
```
Comment (alexander-schranz) [inline src/Sulu/Component/Rest/ListBuilder/Doctrine/FieldDescriptor/DoctrineFieldDescriptor.php]: Did get a interface error in the IDE should we set it to the FieldDescriptorInterface or let it on self

File: UPGRADE.md [hunk 2]
```diff
@@ -2,6 +2,86 @@
...
+
+The field descriptor parameter `$default` and `$disabled` where combined into a new parameter `$visibility`:
+Use the following table for upgrading:
+
+| Disabled | Default  | Visiblity
+|----------|----------|--------------
+| true     | true     | FieldDescriptorInterface::VISIBILITY_ALWAYS (always)
+| true     | false    | FieldDescriptorInterface::VISIBILITY_NEVER (never)
+| false    | true     | FieldDescriptorInterface::VISIBILITY_YES (yes)
+| false    | false    | FieldDescriptorInterface::VISIBILITY_NO (no)
```
[the remark this answers is not in the document]
Comment (alexander-schranz) [reply UPGRADE.md]: @trickreich did you check this?

[the remark this answers is not in the document]
Comment (trickreich) [reply UPGRADE.md]: Sorry.. haven't submitted the form.
I think it should be this way:

| Disabled | Default  | Visiblity
|----------|----------|--------------
| false    | true     | FieldDescriptorInterface::VISIBILITY_ALWAYS (always)
| true     | true     | FieldDescriptorInterface::VISIBILITY_YES (yes)
| true     | false    | FieldDescriptorInterface::VISIBILITY_NO (no)
| false    | false    | FieldDescriptorInterface::VISIBILITY_NEVER (never)

**Comment by danrot** [comment]
@alexander-schranz LGTM, could you rebase so that we can merge it?

File: UPGRADE.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexander-schranz) [reply UPGRADE.md]: @trickreich you sure I think disabled means you can't change it in the overlay and so always and never should be true?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -36,3 +36,4 @@ Describe BC breaks/deprecations here. (remove this section if not needed)
 #### To Do
 
 - [ ] Create a documentation PR
+- [ ] Add UPGRADE
```
Comment (danrot) [inline .github/PULL_REQUEST_TEMPLATE.md]: `Add breaking changes to UPGRADE.md`?

File: UPGRADE.md [hunk 4]
```diff
@@ -19,6 +19,86 @@ doctrine:
...
+
+The field descriptor parameter `$default` and `$disabled` where combined into a new parameter `$visibility`:
+Use the following table for upgrading:
+
+| Disabled | Default  | Visiblity
+|----------|----------|--------------
+| false    | true     | FieldDescriptorInterface::VISIBILITY_ALWAYS (always)
+| false    | false    | FieldDescriptorInterface::VISIBILITY_NEVER (never)
+| true     | true     | FieldDescriptorInterface::VISIBILITY_YES (yes)
+| true     | false    | FieldDescriptorInterface::VISIBILITY_NO (no)
```
Comment (alexander-schranz) [inline UPGRADE.md]: this list was false and was fixed in: https://github.com/sulu/sulu/pull/3991/files

**Pull request merged**